### PR TITLE
Alee/dcs 403 refactor update draft

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,7 +75,7 @@
   },
   "husky": {
     "hooks": {
-      "pre-commit": "lint-staged && tsc && npm test"
```
Comment (andrewrlee) [inline package.json:78]: removing tests as takes >25seconds

File: server/routes/createReport.ts [hunk 2]
```diff
@@ -66,30 +66,15 @@ export default class CreateReport {
...
-      })
-
-      if (updatedPayload) {
-        await this.draftReportService.update({
-          currentUser: res.locals.user,
-          formId,
-          bookingId: parseInt(bookingId, 10),
-          formObject: updatedPayload || {},
-        })
-      }
```
Comment (andrewrlee) [inline server/routes/createReport.ts:92]: pushing this logic down (and also duplicated logic in incident details) into service layer

File: server/services/report/draftReportService.ts [hunk 3]
```diff
@@ -27,8 +27,14 @@ export default class DraftReportService {
...
-  public update(params: UpdateParams): Promise<number> {
-    return this.updateDraftReport.update(params)
+  public process(
+    currentUser: LoggedInUser,
+    bookingId: number,
+    formName: string,
+    updatedSection: unknown,
+    incidentDate?: Date | null
+  ): Promise<void> {
+    return this.updateDraftReport.process(currentUser, bookingId, formName, updatedSection, incidentDate)
```
Comment (andrewrlee) [inline server/services/report/draftReportService.ts:37]: this just delegates down - not sure if there is a way of defining the args as a type so don't have to duplicate, something like: 


public process(
    currentUser: LoggedInUser,
    bookingId: number,
    formName: string,
    updatedSection: unknown,
    incidentDate?: Date | null
  ): Promise<void> {
    return this.updateDraftReport.process(currentUser, bookingId, formName, updatedSection, incidentDate)
 

 
this just delegates down - not sure if there is a way of defining the args as a type so don't have to duplicate, something like a typesafe version of: 
```ts
  public process(
    ...args: UpdateParams
  ): Promise<void> {
    return this.updateDraftReport.process(...args)
}
```

File: server/services/validation/formProcessing.js [hunk 4]
```diff
@@ -9,25 +8,6 @@ const processInput = ({ validationSpec, input }) => {
   return { payloadFields, extractedFields, errors }
 }
 
-/**
- * Create new object from formObject with field 'formName' replaced by formPayload.
```
Comment (andrewrlee) [inline server/services/validation/formProcessing.js:13]: this is now called `getUpdatedReport` and in the draft service

File: server/types/express.d.ts [hunk 5]
```diff
@@ -5,17 +5,27 @@ declare namespace Express {
       token: string
       refreshToken: string
       refreshTime: any
-      firstName: string
-      lastName: string
-      userId: string
-      displayName: string
-      isReviewer: boolean
-      isCoordinator: boolean
-      activeCaseLoadId: string
```
Comment (andrewrlee) [inline server/types/express.d.ts:14]: all this was on `res.locals.user`. There is a cut down version of the user on `req.user`

File: server/services/report/draftReportService.ts (hunk 3 above)
Comment (andrewrlee) [reply server/services/report/draftReportService.ts:37]: might be able to do this in [typescript 3 actually](https://auth0.com/blog/typescript-3-exploring-tuples-the-unknown-type/#:~:text=A%20tuple%20is%20a%20TypeScript,need%20not%20be%20the%20same).

**Review by bell-pa — APPROVED** [review]
LGTM

**Pull request merged**